SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/SpatialDynamicComponentsFastReadditionTest.cpp

```diff
@@ -0,0 +1,97 @@
+﻿#include "SpatialGDK/DynamicSubobjectsTest/SpatialDynamicComponentsFastReadditionTest.h"
+
+#include "Algo/Count.h"
+#include "Engine/World.h"
+#include "EngineUtils.h"
+
+ADynamicComponentsTestActor::ADynamicComponentsTestActor()
+{
+	bReplicates = true;
+	bAlwaysRelevant = true;
+}
+
+USelfRecreatingDynamicComponent::USelfRecreatingDynamicComponent()
+{
+	SetIsReplicatedByDefault(true);
+}
+
+void USelfRecreatingDynamicComponent::PreReplication(IRepChangedPropertyTracker& ChangedPropertyTracker)
+{
+	Super::PreReplication(ChangedPropertyTracker);
+	if (!bNeedsToRecreateItself)
+	{
+		return;
+	}
+
+	bNeedsToRecreateItself = false;
+
+	TSharedPtr<FDelegateHandle> DelegateHandle = MakeShared<FDelegateHandle>();
+	auto OnFrame = [this, DelegateHandle] {
+		// Set another component up.
+		USelfRecreatingDynamicComponent* Another =
+			NewObject<USelfRecreatingDynamicComponent>(GetOwner(), *FString::Printf(TEXT("Recreated")));
+		Another->bNeedsToRecreateItself = false;
+
+		Another->RegisterComponent();
+
+		// Destroy self.
+		DestroyComponent();
+
+		FCoreDelegates::OnEndFrame.Remove(*DelegateHandle);
+	};
+	*DelegateHandle = FCoreDelegates::OnEndFrame.Add(FSimpleDelegate::CreateLambda(OnFrame));
+}
```

Comment:
personally would prefer these to be under the test constructor and preparetest as I read those first

Reply:
I like the reasoning, but would like to get a second opinion from @mironec or @simonsarginson as in my C++ experience, implementations of these kinds of auxiliary classes often goes above the "meat" of the class.